### PR TITLE
feat(core): (de-)serialize documents

### PR DESCRIPTION
**Pull request by inyono**
### Added

- Handle serialization and deserialization of documents

### CODE REVIEW

File: packages/core/src/document/index.tsx [hunk 1]
```diff
@@ -27,7 +27,9 @@ export interface DocumentProps {
 }
 
 export function isDocumentIdentifier(
-  state: DocumentProps['state']
+  state: unknown
 ): state is DocumentIdentifier {
-  return (state as DocumentIdentifier).$$typeof !== undefined
+  return (
+    state !== undefined && (state as DocumentIdentifier).$$typeof !== undefined
```
Comment (Knorrke) [inline packages/core/src/document/index.tsx]: why was this change necessary?

File: packages/core/src/document/index.tsx [hunk 2]
```diff
@@ -27,7 +27,9 @@ export interface DocumentProps {
 }
 
 export function isDocumentIdentifier(
-  state: DocumentProps['state']
+  state: unknown
```
Comment (Knorrke) [inline packages/core/src/document/index.tsx]: why was this change necessary?

Comment (inyono) [reply packages/core/src/document/index.tsx]: So you can also apply this to `null` values.

File: packages/core/src/document/index.tsx (hunk 1 above)
Comment (inyono) [reply packages/core/src/document/index.tsx]: So you can also apply this to `null` values.

File: packages/core/__tests__/index.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  const document = getDocument(store.state, state.id)
+
+  if (!document) {
+    throw new Error('document not found')
+  }
+
+  expect(document.plugin).toEqual('stateless')
+})
+
+test('deserialize nested', () => {
```
Comment (inyono) [inline packages/core/__tests__/index.tsx]: @Knorrke Tests work now as we intended them to do 👍
